File: codes/cpp/chap01/answer_A03.cpp
#include <iostream>
using namespace std;

int N, K;
int P[109], Q[109];
bool Answer = false;

int main() {
	// 입력
	cin >> N >> K;
	for (int i = 1; i <= N; i++) cin >> P[i];
	for (int i = 1; i <= N; i++) cin >> Q[i];

	// 완전 탐색(Answer는 '합계가 K가 되는 선택 방법을 발견했는가'를 나타냄)
	for (int x = 1; x <= N; x++) {
		for (int y = 1; y <= N; y++) {
			if (P[x] + Q[y] == K) Answer = true;
		}
	}

	// 출력
	if (Answer == true) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}

File: codes/python/chap01/answer_A03.py
# 입력
N, K = map(int, input().split())
P = list(map(int, input().split()))
Q = list(map(int, input().split()))
Answer = False

# 완전 탐색(Answer는 '합계가 K가 되는 선택 방법을 발견했는가'를 나타냄)
for x in range(N):
	for y in range(N):
		if P[x] + Q[y] == K:
			Answer = True

# 출력
if Answer == True:
	print("Yes")
else:
	print("No")
